core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/signer/RegionSet.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.signer;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * This class represents the concept of a set of regions.
+ * <p>
+ * A Re
+ * <p>
+ * Examples of a valid region set:
+ * <ul>
+ *     <li>'*' - Represents all regions, i.e. a global scope</li>
+ *     <li>'us-west-2,us-east-1' - Represents 2 regions, us-west-2 and us-east-1</li>
+ *     <li>'eu-west-1' - Represents a single region, eu-west-1</li>
+ * </ul>
+ * <p>
+ * Example of an invalid region set:
+ * <ul>
+ *     <li>'us-*-1,eu-west-1' - A wildcard must be its own item.</li>
+ * </ul>
+ */
+@SdkPublicApi
+public final class RegionSet {
+
+    public static final RegionSet GLOBAL;
+
+    private static final Pattern REGION_SCOPE_PATTERN;
+
+    static {
+        REGION_SCOPE_PATTERN = Pattern.compile("^([a-z0-9-])*|([*])$");
+        GLOBAL = create("*");
+    }
+
+    private final String regionSet;
+
+    private RegionSet(String regionSet) {
+        this.regionSet = Validate.paramNotBlank(regionSet, "regionSet");
+        validateFormat(regionSet);
+    }
+
+    /**
+     * Gets the string representation of this RegionSet.
+     */
+    @Override
+    public String toString() {
```

Comment:
I think we should have a separate method, like `value()` to retrieve the string value, which is used in business logic. Relying to toString() is fragile as someone looking at this class may not realize changing the implementation can be breaking, say we add some other property to this class, one may think of updating toString(), and probably not easy to find all places that may be calling toString() to understand impact. Unlikely that attempting to change the impl here would be missed as it would be caught in integ testing, but generally shouldn't rely on toString().

Reply:
Looks like RegionScope used `id()` for this, so probably just use the same here.